Realm/Tests/RLMAssertions.h

```diff
@@ -63,24 +63,26 @@
 })
 
-#define RLMValidateRealmError(macro_error, macro_errnum, macro_description, macro_underlying)        \
-({                                                                                                   \
-    NSString *macro_dsc = macro_description;                                                         \
-    NSString *macro_usl = macro_underlying;                                                          \
-    NSError *macro_castErr = (NSError *)macro_error;                                                 \
-    XCTAssertNotNil(macro_castErr);                                                                  \
-    XCTAssertEqual(macro_castErr.domain, RLMErrorDomain);                                            \
-    XCTAssertEqual(macro_castErr.code, macro_errnum);                                                \
-    if (macro_dsc.length) {                                                                          \
-        NSString *macro_dscActual = macro_castErr.userInfo[NSLocalizedDescriptionKey];               \
-        XCTAssertNotNil(macro_dscActual);                                                            \
-        XCTAssert([macro_dscActual rangeOfString:macro_dsc].location != NSNotFound);                 \
-    }                                                                                                \
-    if (macro_usl.length) {                                                                          \
-        NSString *macro_uslActual = macro_castErr.userInfo[@"Underlying"];                           \
-        XCTAssertNotNil(macro_uslActual);                                                            \
-        XCTAssert([macro_uslActual rangeOfString:macro_usl].location != NSNotFound);                 \
-    }                                                                                                \
+#define RLMValidateRealmError(macro_error, macro_errnum, macro_description, macro_underlying)            \
+({                                                                                                       \
+    NSString *macro_dsc = macro_description;                                                             \
+    NSString *macro_usl = macro_underlying;                                                              \
+    macro_dsc = [macro_dsc lowercaseString];                                                             \
+    macro_usl = [macro_usl lowercaseString];                                                             \
+    NSError *macro_castErr = (NSError *)macro_error;                                                     \
+    XCTAssertNotNil(macro_castErr);                                                                      \
+    XCTAssertEqual(macro_castErr.domain, RLMErrorDomain, @"Was expecting the error domain '%@', but got '%@' instead", RLMErrorDomain, macro_castErr.domain); \
```

Comment:
I fixed this for the error number test. I decided to keep it for the domain test because this test checks for whether or not the interned copy of the error domain is being used, as well as whether it says the right thing.